eth/options.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Celer Network
+
+package eth
+
+import (
+	"math/big"
+	"time"
+)
+
+type txOptions struct {
+	ethValue           *big.Int      // transact
+	minGasGwei         uint64        // transact
+	maxGasGwei         uint64        // transact
+	addGasGwei         uint64        // transact
+	gasLimit           uint64        // transact
+	blockDelay         uint64        // waitMined
+	pollingInterval    time.Duration // waitMined
+	timeout            time.Duration // waitMined
+	queryTimeout       time.Duration // waitMined
+	queryRetryInterval time.Duration // waitMined
+}
+
+const (
+	defaultPollingInterval      = 15 * time.Second
+	defaultTxTimeout            = 6 * time.Hour
+	defaultTxQueryTimeout       = 2 * time.Minute
+	defaultTxQueryRetryInterval = 10 * time.Second
+)
+
+func defaultTxOptions() txOptions {
```

Comment:
nit: It's more "go" to return "*txOptions" and it seems only its pointer is used below

Reply:
I was using grpc-go as a reference: https://github.com/grpc/grpc-go/blob/v1.29.1/dialoptions.go#L562

One thing special there is that we always use deep copy when using the default options in our code, e.g., https://github.com/celer-network/goutils/blob/f9b58745ee/eth/transactor.go#L116-L119

Reply:
Ah, now I see as I read more code, it's always modifying a new fresh code.  Makes sense. thanks.

Reply:
done